### ISSUE TITLE
user facingMode without mirror effect iOS 12.1.1

### ISSUE THREAD

**Issue by hoodsey20** [JodusNodus/react-qr-reader#99]
## Expected behavior:
Video stream reflected (like in the mirror)

## Observed behavior:
The QR-reader displays video from front-camera without mirror-effect

## Tested on:
iPad iOS 12.1.1 (Safari)
package version 2.1.1

**Comment by jteich** [comment]
I have observed this behavior on all Windows and Chromebook devices that I have tested. The issue is that the code uses a regex string-match to determine if the camera is a user-facing camera. This works on my macbook where the camere is named "Facetime HD camera". On other devices, the name cannot be used to determine if it is an user-facing camera. 
There are a couple of options for addressing this. 

1. enumerate devices with "navigator.mediaDevices.enumerateDevices()" and if only a single device video device is returned, assume that it is user facing. ( similar to the approach discussed in https://github.com/JodusNodus/react-qr-reader/issues/46 ). 

2. Test with both the current front-facing and rear-facing regexes, and in the event that neither match (or that both match), assume that it is user-facing. 

3. Same as #1, but allow the assumption of user or back-facing to be configurable

4. Same as #2, but allow the assumption of user or back-facing to be configurable

5. Disable facing-detection, and just allow the mirroring to be configurable.  

6. Disable facing-detection and just use the "facingMode" property (and use a configurable default if it isn't specified). 

7. Have a property that takes a closure that takes the camera name and how many other video devices there on and returns a value to determine if mirroring should be enabled.  

8. Only enable facing-mode detection by regex if the camera name is a known camera where it the facing-orientation is deterministic. In this case, there will still need to a (configurable) default. 
 
I am right now leaning towards #6. I think that the facingMode detection by pattern match is shakey. It could be done more deterministically but checking the UA, but it's still going to break in many cases. 

Additionally, the onLoad callback should have an object as an argument that specifies the name of the camera and the facing-mode in case the application using this needs to know in order to present the user with a UI to flip mirroring.
